picatrix/lib/manager.py

```diff
@@ -142,4 +144,14 @@ def register_magic(
     cls._magics[magic_name] = function
     function_name = f'{magic_name}_func'
 
-    _ = utils.ipython_bind_global(function_name, function.fn)
+    def capture_output(function, name):
```

Comment:
Not overly important, but consider that you could hand over the state as an argument here and let the caller deal with getting the state:

```
def capture_output(state, function, name):
  ...

_ = utils.ipython_bind_global(
        function_name, capture_output(state.get_state(), function.fn, function_name))
```

Perhaps makes it easier to test the function.

Could even add state to the MagicManager at init time so the methods can refer to that state rather than reaching out to the global one.